vignettes/trial_analysis.Rmd

```diff
@@ -0,0 +1,606 @@
+
+#### The stopping rule
+The trial will stop when <i>either</i>
+
+* Twenty patients have been recruited, or.
+* Both of the following conditions are true
+  * At least three cohorts must have been treated AND
+  * The probability that the current estimate of the MTD is in the target toxicity range must be at least 0.5.
+
+
+#### Trial definition
+The code to define a trial with this design is
+```{r}
+doseGrid <- c(1, 3, 9, 20, 30, 45, 60, 80, 100)
+empty_data <- Data(doseGrid = doseGrid)
+
+model <- LogisticLogNormal(
+           mean = c(-0.85, 1),
+           cov = matrix(c(1, -0.5, -0.5, 1), nrow = 2),
+           refDose = 56
+         )
+
+my_increments <- IncrementsRelative(
+                   intervals = c(0, 30),
+                   increments = c(1, 0.5)
+                 )
+
+my_next_best <- NextBestNCRM(
+                  target = c(0.2, 0.35),
+                  overdose = c(0.35, 1),
+                  maxOverdoseProb = 0.25
+                )
+
+my_size <- maxSize(
+             CohortSizeRange(intervals = c(0, 30), cohortSize = c(1, 3)), 
+             CohortSizeDLT(DLTintervals = c(0, 1), cohortSize = c(1, 3))
+           )
+
+my_stopping <- (StoppingMinCohorts(nCohorts = 3) & 
+                StoppingTargetProb(target = c(0.2, 0.35), prob = 0.5)) |
+               StoppingMinPatients(nPatients = 20)
+
+
+design <- Design(
+  model = model,
+  nextBest = my_next_best,
+  stopping = my_stopping,
+  increments = my_increments,
+  cohortSize = my_size,
+  data = empty_data,
+  startingDose = 3
+)
+```
+
+## Analysing a trial
+Given the trial design constructed above, the process of analysing a real life instance of the trial is simply a matter of providing the model with the actual toxicity status of the participants treated so far.  The escalation rules we defined earlier allow the use of a single patient run-in until either the first DLT is observed or until dose 20 has been administered.  
+
+### The single patient run-in
+Assume that the first three patients - dosed at `1`, `3` and `5` - completed the trial without incident, but that the fourth patient - treated at `10` - experienced a DLT.
+
+We provide this information to `crmPack` via a `Data` object:
+
+```{r}
+firstFour <- Data(
+              x=c(1, 3, 9, 20),
+              y=c(0, 0, 0,  1),
+              ID=1:4,
+              cohort=1:4,
+              doseGrid=doseGrid
+             )
+```
+Within a `Data` object, the doses at which each patient is treated are given by the `x` slot and their toxicity status (a Boolean where a toxicity is represented by a truthy value) by the `y` slot.
+
+The observed data is easily visualised
+```{r}
+plot(firstFour)
+```
+
+and, since the `plot` method returns a `ggplot` object, it is easily customised.
+
+```{r}
+plot(firstFour) + theme_light()
+```
+
+Now, update the model to obtain the posterior estimate of the dose-toxicty curve:
```

Comment:
```suggestion
Now, update the model to obtain the posterior estimate of the dose-toxicity curve:
```
you can run spell check as well or see the spelling errors reported in the github action checks